app/components/verkiezingen/prepare-legislatuur-section.js

```diff
@@ -32,17 +43,177 @@ export default class PrepareLegislatuurSectionComponent extends Component {
+      const mandatenOpBestuursOrgaan = await this.args.bestuursorgaan.bevat;
+      let mandaat = null;
+
+      for (const existingMandaat of mandatenOpBestuursOrgaan) {
+        if (mandaat) {
+          continue;
+        }
+        const bestuursfunctie = await existingMandaat.bestuursfunctie;
+
+        if (bestuursfunctie.uri == toBestuursfunctie.uri) {
+          mandaat = existingMandaat;
+        }
+      }
+
+      if (!mandaat) {
+        mandaat = this.store.createRecord('mandaat', {
+          bestuursfunctie: toBestuursfunctie,
+          bevatIn: [this.args.bestuursorgaan],
+        });
+        await mandaat.save();
+      }
+
+      const lidmaatschap = await mandatarisToAdd.heeftLidmaatschap;
+      let newLidmaatschap = null;
+      if (lidmaatschap) {
+        newLidmaatschap = this.store.createRecord('lidmaatschap', {
+          binnenFractie: await lidmaatschap.binnenFractie,
+          lidGedurende: await lidmaatschap.lidGedurende,
+        });
+      }
+
+      const newMandataris = await this.createMandatarisFromMandataris(
+        mandatarisToAdd,
+        mandaat,
+        newLidmaatschap
+      );
+
+      if (newLidmaatschap) {
+        newLidmaatschap.lid = newMandataris;
+        await newLidmaatschap.save();
+      }
+
+      const mandatarissen = await mandaat.bekleedDoor;
+      mandatarissen.pushObject(newMandataris);
+      await mandatarissen.save();
+    }
+  });
+
+  async createMandatarisFromMandataris(mandataris, mandaat, lidmaatschap) {
+    return this.store.createRecord('mandataris', {
+      rangorde: mandataris.rangorde,
+      start: mandataris.start,
+      einde: mandataris.einde,
+      bekleedt: mandaat,
+      isBestuurlijkeAliasVan: await mandataris.isBestuurlijkeAliasVan,
+      beleidsdomein: await mandataris.beleidsdomein,
+      status: await getEffectiefStatus(this.store),
+      publicationStatus: await getDraftPublicationStatus(this.store),
+      heeftLidmaatschap: lidmaatschap,
+    });
+  }
+
+  async getBestuursorgaanMandatarissen(bestuursorgaan) {
+    const queryParams = {
+      page: {
+        number: 0,
+        size: 9999,
+      },
+      filter: {
+        bekleedt: {
+          'bevat-in': {
+            id: bestuursorgaan.id,
+          },
+        },
+      },
+      include: [
+        'is-bestuurlijke-alias-van',
+        'bekleedt.bestuursfunctie',
+        'heeft-lidmaatschap.binnen-fractie',
+        'status',
+        'beleidsdomein',
+      ].join(','),
+    };
+
+    return await this.store.query('mandataris', queryParams);
+  }
+
   @action
   cancel() {
     this.editMode = null;
   }
 
-  @action
-  async onCreate({ instanceTtl, instanceId }) {
+  onCreate = restartableTask(async ({ instanceTtl, instanceId }) => {
     this.editMode = null;
     await syncNewMandatarisMembership(this.store, instanceTtl, instanceId);
-    setTimeout(() => this.router.refresh(), 1000);
-  }
+    await timeout(1000);
+  });
```

Comment:
Changed it that it will not refresh the whole page when adding an item but just refreshes the table